scipy/sparse/_construct.py

```diff
@@ -1233,7 +1238,7 @@ def data_sampler(size):
             data_sampler = rng.uniform
 
     # rng.choice uses int64 if first arg is an int
-    if tot_prod < np.iinfo(np.int64).max:
+    if tot_prod <= np.iinfo(np.int64).max:
```

Comment:
This is unrelated to the others, but I noticed it because I was looking at use of `np.iinfo`.  `<` should be `<=`
Not gonna be noticed, especially because there is a workaround in the else clause. But in case someoen is **really** pushing the boundaries of int64. :)